build/build.fsx

```diff
@@ -1,20 +1,64 @@
-#r @"../packages/FAKE.4.60.0/tools/FakeLib.dll"
+#r @"../packages/FAKE.4.63.0/tools/FakeLib.dll"
+#load @"ExtractDocs.fsx"
 
 open Fake
+open Fake.TargetHelper
 open System
+open System.IO
+open System.Text.RegularExpressions
+open ExtractDocs
 
-let configuration = getBuildParamOrDefault "configuration" "Debug"
+module FileReaderWriter =
+    let Read file = File.ReadAllText(file)
+    let Write file text = File.WriteAllText(file, text)
+    let TransformFile file target (f : string -> string) =
+        Read file
+        |> f
+        |> Write target
+
+module ExamplesToCode =
+    open FileReaderWriter
+
+    let ConvertFile file targetDir =
+        let fileName = Path.GetFileNameWithoutExtension(file)
+        let target = targetDir @@ fileName + ".cs"
+        log <| sprintf "Converting %s to %s" file target
+        TransformFile file target (ExtractDocs.strToFixture fileName)
+
+    let Convert paths targetDir =
+        let paths = paths |> Seq.toList
+        for p in paths do
+            trace <| sprintf "Convert from %s to %s" p targetDir
+            let files = !! "*.markdown" ++ "*.html" |> SetBaseDir p
+            for file in files do
+                ConvertFile file targetDir
+
+let getVersion () =
+    let tag = Git.CommandHelper.runSimpleGitCommand "" "describe --tags --long"
+    let result = Regex.Match(tag, @"(v|alpha|beta|rc)(\d+)\.(\d+)\.(\d+)\-(\d+)").Groups
+    let getMatch (i:int) = result.[i].Value
+    (sprintf "%s.%s.%s.%s" (getMatch 2) (getMatch 3) (getMatch 4) (getMatch 5), match getMatch 1 with "v" -> "" | suffix -> suffix)
```

Comment:
Why not use a tool like [GitVersion](https://github.com/GitTools/GitVersion) for versioning? Not sure if it works well with FAKE.

Reply:
FAKE has a helper for GitVersion https://fake.build/apidocs/fake-gitversionhelper.html
I prefer to have something like this https://github.com/dwijnand/sbt-dynver#detail

Reply:
:+1:

Reply:
Didn't know about GitVersion, thanks! I'd like to avoid needing another dependency though. Also I like the current hackish code gets a new version number each commit.